games-fps/serioussam/serioussam-1.10.6.ebuild

```
@@ -0,0 +1,151 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake desktop
+
+MY_PN1="SamTFE"
+MY_PN2="SamTSE"
+# Game name
+GN1="serioussam"
+GN2="serioussamse"
+
+DESCRIPTION="Linux port of Serious Sam Classic with Vulkan support"
+HOMEPAGE="https://github.com/tx00100xt/SeriousSamClassic-VK"
+SRC_URI="https://github.com/tx00100xt/SeriousSamClassic-VK/archive/refs/tags/${PV}c.tar.gz -> ${P}c.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+RESTRICT="bindist mirror"
+IUSE="alsa pipewire vulkan"
+
+RDEPEND="
+	media-libs/libsdl2[video,joystick,opengl]
+	media-libs/libvorbis
+	sys-libs/zlib
+	alsa? (
+		>=media-libs/libsdl2-2.0.6[alsa,sound]
+	)
+	pipewire? (
+		>=media-libs/libsdl2-2.0.6[pipewire,sound]
+	)
+	vulkan? (
+		dev-util/vulkan-headers
+		media-libs/vulkan-loader
+		media-libs/libsdl2[video,joystick,opengl,vulkan]
+	)
+"
+
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	sys-devel/flex
+	sys-devel/bison
+	media-gfx/imagemagick
+"
+
+S="${WORKDIR}/SeriousSamClassic-VK-${PV}c"
+MY_CONTENT1="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN1}"
+MY_CONTENT2="${WORKDIR}/SeriousSamClassic-VK-${PV}c/${MY_PN2}"
+BUILD_TMP=${BUILD_DIR}
+
+src_configure() {
+	einfo "Remove Win32 stuff..."
+	rm -rf "${MY_CONTENT1}"/Tools.Win32 || die "Failed to removed Win32 stuff"
+
+	einfo "Setting build type Release..."
+	CMAKE_BUILD_TYPE="Release"
+	local mycmakeargs=(
+		-DUSE_SYSTEM_INSTALL=ON
+	)
+	if use vulkan; then
+		mycmakeargs+=(
+			-DUSE_SYSTEM_VULKAN=ON
+		)
+	else
+		mycmakeargs+=(
+			-DUSE_SYSTEM_VULKAN=OFF
+		)
+	fi
+	if use x86; then
+		mycmakeargs+=(
+			-DUSE_ASM=OFF
+		)
+	fi
```

i think you could simplify this a bit:
```suggestion
	local mycmakeargs=(
		-DUSE_SYSTEM_INSTALL=ON
		-DUSE_SYSTEM_VULKAN=$(usex vulkan ON OFF)
		-DUSE_ASM=$(usex x86 OFF ON)
	)
```
Haven't tested this so please check if it works .

---

Works. Tested x86 and amd64.
Everything works correctly.